src/legacy/ui/public/new_platform/new_platform.karma_mock.js

```diff
@@ -154,6 +154,9 @@ export const npSetup = {
         }),
       },
     },
+    spaces: {
```

Comment:
It was a while ago..I had Karma test failures, so I followed the instructions in the migration guide to "fix" it. I have a lot less in the LP plugin now though, so maybe I'm already past the issue. I'll remove and see what happens. If it fails, I should be able to introduce a mock within the spaces plugin itself.

https://github.com/elastic/kibana/blob/5d6dbf07b028647ee251601a56ef4fb74fb2506c/src/core/MIGRATION.md#L1564-L1570

Reply:
Update: the LP spaces plugin runs it's [`legacy.ts`](https://github.com/elastic/kibana/blob/dca42cc6830e1e52addae4008e6a6cbcef66657c/x-pack/legacy/plugins/spaces/public/legacy.ts) as part of a `hack`, so that it runs for all LP plugins. The goal of `legacy.ts` is to call the NP `registerLegacyAPI`, so that the plugin can finish its initialization. In the case of these Karma tests, we need the `npSetup` static import provided by `new_platform.karma.mock` to have the `spaces` plugin registered, so that `legacy.ts` can call `registerLegacyAPI`.

So an alternative to placing this here, is to place a guard within `legacy.ts`, to only call `registerLegacyAPI` if the `spaces` plugin is actually defined. What do you think about that approach?

Reply:
> So an alternative to placing this here, is to place a guard within legacy.ts, to only call registerLegacyAPI if the spaces plugin is actually defined. What do you think about that approach?

On the surface it sounds like a better workaround to me, do you have any concerns regarding it?

Reply:
And thanks for the explanation!

Reply:
> On the surface it sounds like a better workaround to me, do you have any concerns regarding it?

None at all! Resolved in https://github.com/elastic/kibana/pull/54298/commits/3f249027b93f9540958bb3ebce00164176e3c1c1